File: kmp/composeApp/src/commonMain/kotlin/dev/jianastrero/trainer/ui/theme/Color.kt
package dev.jianastrero.trainer.ui.theme

import androidx.compose.material.Colors
import androidx.compose.ui.graphics.Color

val Primary = Color(0xFFE53935)
val Light = Color(0xFFECEFF1)
val Dark = Color(0xFF2C3E50)
val Error = Color(0xFFB00020)

val TrainerLightColors = Colors(
    primary = Primary,
    primaryVariant = Primary,
    secondary = Primary,
    secondaryVariant = Primary,
    background = Light,
    surface = Light,
    error = Error,
    onPrimary = Light,
    onSecondary = Light,
    onBackground = Dark,
    onSurface = Dark,
    onError = Light,
    isLight = true
)

val TrainerDarkColors = Colors(
    primary = Primary,
    primaryVariant = Primary,
    secondary = Primary,
    secondaryVariant = Primary,
    background = Dark,
    surface = Dark,
    error = Error,
    onPrimary = Light,
    onSecondary = Light,
    onBackground = Light,
    onSurface = Light,
    onError = Light,
    isLight = false
)

File: react_native/src/ui/theme/Color.ts
const PrimaryColor = '#E53935'
const LightColor = '#ECEFF1'
const DarkColor = '#2C3E50'
const ErrorColor = '#B00020'

export interface Colors {
    primary: string;
    primaryVariant: string;
    secondary: string;
    secondaryVariant: string;
    background: string;
    surface: string;
    error: string;
    onPrimary: string;
    onSecondary: string;
    onBackground: string;
    onSurface: string;
    onError: string;
}

export const TrainerLightColors: Colors = {
    primary: PrimaryColor,
    primaryVariant: PrimaryColor,
    secondary: PrimaryColor,
    secondaryVariant: PrimaryColor,
    background: LightColor,
    surface: LightColor,
    error: ErrorColor,
    onPrimary: LightColor,
    onSecondary: LightColor,
    onBackground: DarkColor,
    onSurface: DarkColor,
    onError: LightColor,
};

export const TrainerDarkColors: Colors = {
    primary: PrimaryColor,
    primaryVariant: PrimaryColor,
    secondary: PrimaryColor,
    secondaryVariant: PrimaryColor,
    background: DarkColor,
    surface: DarkColor,
    error: ErrorColor,
    onPrimary: LightColor,
    onSecondary: LightColor,
    onBackground: LightColor,
    onSurface: LightColor,
    onError: LightColor,
};
